AUE 388

jweible opened the pull request:
Example output:

Analysis Project: d98503f8cc9e49c1a3c9b6a30c94328d
Name: CLE Validation AML Pairs genome-3551
Status: Hold
Run as: cle-staging
Created At: 2014-12-07 13:49:39
Updated: 2014-12-18 10:29:37
Created By: jwalker

Subject Mapping: 0939546b37944677b6f2ec548aff557d
    Subject: 05eda9363b4c40599375b2a8fa476462
        Subject: 2811093070
        Label: tumor_sample
        Created By: jwalker
        Created At: 2014-12-07 13:59:35
        Updated At: 2014-12-07 13:59:35
        Tags: discovery

    Subject: 4db5235abaf34649b2a14876e21e6ed5...

tmooney commented on lib/perl/Genome/Config/AnalysisProject/SubjectMapping/Command/View.pm:
```diff
@@ -0,0 +1,149 @@
+    );
+}
+
+sub _write_pairs_line {
```
It looks like `_write_pairs_line` was copied verbatim from the AnalysisProject view command.  Could this be pulled up into a parent class?

tmooney commented:
It might be good to add a small test that sets up a test AnP and some subject mappings to make sure the command runs to completion.

acoffman commented:
Personal preference, but I believe that most people specify their subject mappings on import based on name, so I wonder if it would be good to show the name here? That would slow the command down a bit, especially for large AnPs so maybe it should be an option? Or maybe a UR hint would help? I'm not sure exactly how that works...

jasonwalker80 commented:
Names would be much more beneficial for the end user over the UUID.

tmooney commented on lib/perl/Genome/Config/AnalysisProject/SubjectMapping/Command/View.pm:
```diff
@@ -0,0 +1,149 @@
+        ['Created By', $subject->created_by], 
+        ['Created At', $subject->created_at], 
+        ['Updated At', $subject->updated_at], 
+        ['Tags', join(',', map{$_->name} $subject->subject_mapping->tags)],
```
If the tags are properties of the mapping, they needn't be repeated with each subject in the mapping.

jweible commented:
Commits added to address feedback.  Current output looks like so:

jweible commented:
Analysis Project: d98503f8cc9e49c1a3c9b6a30c94328d
Name: CLE Validation AML Pairs genome-3551
Status: Hold
Run as: cle-staging
Created At: 2014-12-07 13:49:39
Updated: 2014-12-18 10:29:37
Created By: jwalker

Subject Mapping: 0939546b37944677b6f2ec548aff557d
Tags: discovery
    Subject: H_KA-702808-0909968
        Subject: 2811093070
        Label: tumor_sample
        Created By: jwalker
        Created At: 2014-12-07 13:59:35
        Updated At: 2014-12-07 13:59:35

    Subject: H_KA-702808-0900124
        Subject: 2811097611
        Label: normal_sample
        Created By: jwalker
        Created At: 2014-12-07 13:59:35
        Updated At: 2014-12-07 13:59:35

Subject Mapping: 1187e394a5a246c6b70629788a3343ea
Tags: germline 
...

jweible commented:
Example output (with names):
Analysis Project: d98503f8cc9e49c1a3c9b6a30c94328d
Name: CLE Validation AML Pairs genome-3551
Status: Hold
Run as: cle-staging                      Created By: jwalker
Created: 2014-12-07 13:49:39             Updated: 2014-12-18 10:29:37

=== Subject Mapping (0939546b37944677b6f2ec548aff557d): ===
    Tags: discovery
    Subject: H_KA-702808-0909968 (AML37 tumor 2811093070)
        Label: tumor_sample
        Created By: jwalker
        Created: 2014-12-07 13:59:35             Updated: 2014-12-07 13:59:35

    Subject: H_KA-702808-0900124 (AML37 normal 2811097611)
        Label: normal_sample
        Created By: jwalker
        Created: 2014-12-07 13:59:35             Updated: 2014-12-07 13:59:35

=== Subject Mapping (1187e394a5a246c6b70629788a3343ea): ===
    Tags: germline
    Subject: H_KA-804168-0802136 (AML40 normal 2811092681)
        Label: tumor_sample
        Created By: jwalker
        Created: 2014-12-07 13:59:35             Updated: 2014-12-07 13:59:35

jweible commented:
@tmooney : Duplicate Code removed.  A test that runs use_ok has been added.  I do not believe a test that just runs the command without checking the output is worthwhile, and getting this out has already taken too much time.

jweible commented:
@acoffman & @jasonwalker80: Names are in.  See above for the latest example output

jweible commented:
@tmooney: Acknowledged and fixed

jasonwalker80 commented:
+1

acoffman commented:
+1

tmooney commented:
+1 :facepunch:
